services/etcd.go

```diff
@@ -318,6 +318,47 @@ func RunEtcdSnapshotSave(ctx context.Context, etcdHost *hosts.Host, prsMap map[s
 	return nil
 }
 
+func DownloadEtcdSnapshot(ctx context.Context, etcdHost *hosts.Host, prsMap map[string]v3.PrivateRegistry, etcdSnapshotImage string, name string, once bool, es v3.ETCDService) error {
+	log.Infof(ctx, "[etcd] Get snapshot [%s] on host [%s]", name, etcdHost.Address)
+	s3Backend := es.BackupTarget.S3BackupTarget
+	if len(s3Backend.Endpoint) == 0 || len(s3Backend.AccessKey) == 0 || len(s3Backend.SecretKey) == 0 || len(s3Backend.BucketName) == 0 {
```

Comment:
This looks like it would break, since you don't pass the access/secret key anymore. Also, this would break if you use an IAM role.

Reply:
addressed.